crypto/s2n_ecc_evp.h

```diff
@@ -15,20 +15,33 @@
 
 #pragma once
 
-#include "crypto/s2n_ecc.h"
-
 #include <openssl/evp.h>
 
 #include "crypto/s2n_hash.h"
+#include "tls/s2n_kex_data.h"
 #include "stuffer/s2n_stuffer.h"
 #include "tls/s2n_tls_parameters.h"
 #include "utils/s2n_safety.h"
 
+struct s2n_ecc_named_curve {
+    /* See https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-8 */
+    uint16_t iana_id;
+    /* See nid_list in openssl/ssl/t1_lib.c */
+    int libcrypto_nid;
+    const char *name;
+    const uint8_t share_size;
+};
+
+extern const struct s2n_ecc_named_curve s2n_ecc_curve_secp256r1;
+extern const struct s2n_ecc_named_curve s2n_ecc_curve_secp384r1;
+
 #if S2N_OPENSSL_VERSION_AT_LEAST(1, 1, 0) && !defined(LIBRESSL_VERSION_NUMBER)
 #define MODERN_EC_SUPPORTED 1
+#define S2N_ECC_EVP_SUPPORTED_CURVES_COUNT 3
```

Comment:
Do you still need this `#define`? Can you use `s2n_ecc_evp_supported_curves_list_len` instead?

Reply:
yes, it requires a constant value, and not a value that will be determined at compile time.